### PR TITLE
Path Auto-Keyframe

### PR DESCRIPTION
**Pull request by ghost**
OK to merge.

Keyframe behavior matches everything else when modifying directly-selected shapes.

Summary of changes:

- Enabled numeric parsing for appropriate keyframe values.
- When modifying a path or other directly-selected shape, keyframes are inserted at the current timeline time, just like all other properties.
- When converting a shape with keyframes to a path, corresponding keyframes for the path`d` attribute are created to give the shape the same animation. **Easing is omitted here, because multiple keyframe channels are merged.**

Regressions to look for:

- ?

Completed checkin tasks:

- [x] Did manual testing of interrelated functionality
- [ ] Wrote an automated test for existing and modified functionality
- [ ] Added measurement instrumentation (Mixpanel, etc.)

### CODE REVIEW

File: packages/@haiku/core/src/Transitions.ts [hunk 1]
```diff
@@ -36,7 +23,7 @@ function sortedKeyframes (keyframeGroup) {
 // 2500: { value: { ... } }
 // 5000: { value: { ... } }
 function getKeyframesList (keyframeGroup, nowValue) {
-  const sorted = sortedKeyframesCached(keyframeGroup);
+  const sorted = sortedKeyframes(keyframeGroup);
```
Comment (stristr) [inline packages/@haiku/core/src/Transitions.ts:26]: @winduptoy, what is the rationale for doing away with the concept of the sorted keyframes cache?

I'm not the author of this optimization and I haven't really benchmarked to introspect its significance, but given how hot this code path is, I would expect it to matter at least somewhat.

Comment (tsortwehttam) [reply packages/@haiku/core/src/Transitions.ts:26]: I suggested this. It fixes an implementation detail making it unnecessary to check for the presence of `__sorted` in a key path, something I've wanted to get rid of for a long time. (My rationale is: We are going to do a perf pass next week at which point we can make a more informed decision about how to improve this bottleneck using our more modern caching utilities which are less race-/invalidation-bug-prone (and which weren't around when I added this hack back in the early days).)

Comment (stristr) [reply packages/@haiku/core/src/Transitions.ts:26]: Okay. I would expect this to lower perf (sorting has a cost, and this is essentially triggered on all properties on all mutable timelines on every tick), but agree there's a better way to handle this upstream.

**Comment by stristr** [comment]
LGTM.

**Pull request merged**